src/corecel/data/detail/CollectionImpl.hh

```diff
@@ -198,6 +220,20 @@ struct CollectionAssigner<Ownership::value, MemSpace::device>
     }
 };
 
+//---------------------------------------------------------------------------//
+//! Assignment semantics for copying to mapped memory
+template<>
+struct CollectionAssigner<Ownership::value, MemSpace::mapped>
+{
+    template<class T, Ownership W2, MemSpace M2>
+    auto operator()(CollectionStorage<T, W2, M2> const& source)
+        -> CollectionStorage<T, Ownership::value, M2>
+    {
+        CELER_EXPECT(celeritas::device().support_mapped_memory());
```

Comment:
I think semantically the best place to call `CELER_VALIDATE` would be in the CollectionStorage ctor but it would require quite a bit of refactoring without an actual difference at runtime.